src/brevitas/core/function_wrapper/ops_ste.py

```diff
@@ -38,6 +38,34 @@ def forward(self, x: torch.Tensor):
         return floor_ste(x)
 
 
+class AdaRoundSte(brevitas.jit.ScriptModule):
+    """
+    This Module implements AdaRound representation, where each weight has a learnable parameter
+    that decides if "ceil" or "floor" rounding type has to be used.
+    """
+
+    def __init__(self, shape: int) -> None:
```

Comment:
change to tracked_parameter_list.
Check if tracked_parameter_list is 1, otherwise raise error.
tracked_parameter_list[0].shape